src/nodes.coffee

```diff
@@ -1639,7 +1639,8 @@ exports.Op = class Op extends Base
     not @second
 
   isComplex: ->
-    not (@isUnary() and @operator in ['+', '-']) or @first.isComplex()
+    not (@isUnary() and @operator in ['+', '-'] and
+         @first instanceof Value and @first.isSimpleNumber())
```

Comment:
That's not a very good reason to do an `instanceof` test when we have `isComplex` specifically for this. Which test cases are breaking? Why?